api/agent/agent.go

```diff
@@ -540,19 +546,21 @@ func (s *hotSlot) exec(ctx context.Context, call *call) error {
 
 	select {
 	case err := <-s.errC: // error from container
+		s.trySetError(err)
```

Comment:
this one seems redundant I guess since we're shutting the container slot goroutine down once the container exits ? tiny race b/w doing that I suppose ? this is ok, just curious

Reply:
This is just for the sake of completeness. But in theory, the main runhot go routine may pause after queuing the errC before calling shutdown cancel function. During this time, inner slot processing go routine may have the chance to requeue the slot. But requeue can happen after slot close and there's a lot of code in submit for this to happen, so it's super unlikely.